### PR TITLE
Add RPCs for checking peer best blocks & target blocks

### LINKED ISSUE #71: Add RPCs for checking sync peers & sync target blocks

### PR DESCRIPTION
**Pull request by Byeongjee**
It resolves #71

### CODE REVIEW

File: sync/src/block/extension.rs [hunk 1]
```diff
@@ -663,6 +663,16 @@ impl NetworkExtension<Event> for Extension {
                     channel.send(*peer).unwrap();
                 }
             }
+            Event::GetPeerBestBlockHashes(channel) => {
+                for header_downloader in self.header_downloaders.values() {
+                    channel.send(header_downloader.best_hash()).unwrap();
```
Comment (majecty) [inline sync/src/block/extension.rs]: We need the information which peer has which best block hash.
Please pass the IP address of the peer also.

**Comment by HoOngEe** [comment]
Is there any special reason to select `ipAddr` as a return value? I think ip address is not enough to be used in peer identification. `SocketAddr` seems to be great. @majecty what do you think of it?

**Comment by majecty** [comment]
Aha, SocketAddr has a port number. I agree with @HoOngEe.

**Comment by Byeongjee** [comment]
I changed IpAddr into SocketAddr and updated JSON-RPC documentation

File: spec/JSON-RPC.md [hunk 2]
```diff
@@ -3063,6 +3064,70 @@ No parameters
...
+
+Get IP address and best block hash of each peer.
+
+### Params
+
+No parameters
+
+### Returns
+
+{ SocketAddress: `string`, BlockHash: `H256`}[]
```
Comment (majecty) [inline spec/JSON-RPC.md]: This type does not match the example.
What you said:
```js
[ 
{ "SocketAddress": "xxx", "BlockHash": "yyy"},
{ "SocketAddress": "xxx2", "BlockHash": "yyy2"}, ...
]
```

However, in the example:
```js
[["1.2.3.4:3485", "0x56642f04d519ae3262c7ba6facf1c5b11450ebaeb7955337cfbc45420d573077"]]
```

Comment (Byeongjee) [reply spec/JSON-RPC.md]: Okay.  Should I introduce new type such as SocketAddressAndHash, or tuple type is sufficient?

Comment (majecty) [reply spec/JSON-RPC.md]: IMO, tuple type is sufficient.

Comment (Byeongjee) [reply spec/JSON-RPC.md]: Then the return type should be [ 'string', 'H256'][]?

Comment (majecty) [reply spec/JSON-RPC.md]: It looks good to me.

**Comment by Byeongjee** [comment]
I fixed both license part and return type.

File: rpc/src/v1/traits/chain.rs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018-2019 Kodebox, Inc.
+// Copyright 2018-2020 Kodebox, Inc.
```
Comment (majecty) [inline rpc/src/v1/traits/chain.rs]: Please update the copyrights in the files that you changed.

Comment (Byeongjee) [reply rpc/src/v1/traits/chain.rs]: Sorry. I changed the copyright of wrong file. I pushed the fixed version.

**Comment by Byeongjee** [comment]
There was timeout exceptions in some tests, but they passed successfully in my local machine. @majecty

**Pull request merged**